reference/extensions/deployers.rst

```diff
@@ -1,5 +1,76 @@
 .. _reference_extensions_deployers:
 
 Deployers
----------
+=========
 
+Deployers are a mechanism to facilitate copying files form one folder, usually the Conan cache, to user folders.
+While Conan provides two built-in ones (``full_deploy`` and ``direct_deploy``), users can easily manage their own
+with ``conan config install``.
+
+Deployers run before generators, and they can change the target folders.
+For example, if the ``--deploy=full_deploy`` deployer runs before ``CMakeDeps``,
+the files generated by ``CMakeDeps`` will point to the local copy in the user folder done by the ``full_deploy`` deployer,
+and not to the Conan cache. Multiple deployers can be specified by supplying more than one ``--deploy=`` argument,
+and they will be ran in order of appearance.
+
+Deployers can be multi-configuration. Running ``conan install . --deploy=full_deploy`` repeatedly for different profiles
+can achieve a fully self-contained project, including all the artifacts, binaries, and build files.
+This project will be completely independent of Conan and no longer require it at all to build.
+
+
+Built-in deployers
+------------------
+
+.. _reference_extensions_deployer_full_deploy:
+
+full_deploy()
+^^^^^^^^^^^^^
+
+Deploys each package folder of every dependency to your recipe's ``output_folder`` in a subfolder tree based on:
+
+#. The build context
+#. The dependency name and version
+#. The build type
+#. The build arch
+
+Such that every dependency will end up in a folder such as:
+
+``[OUTPUT_FOLDER]/host/dep/0.1/Release/x86_64``
+
+
+.. _reference_extensions_deployer_direct_deploy:
+
+direct_deploy()
+^^^^^^^^^^^^^^^
+
+Same as ``full_deploy()```, but only processes your recipe's *direct* dependencies.
+
+
+Custom deployers
+----------------
+
+Custom deployers can be managed via ``conan config install``. When looking for a specific deployer,
+Conan will look in these locations for the deployer in the following order:
+
+#. Absolute paths
+#. Relative to cwd
+#. In the ``[CONAN_HOME]/extensions/deploy`` folder
+#. As built-in deployers
+
+For each installed file, Conan will look for a ``deploy()`` method which to call.
+The function signature of your custom deployers should be as follows:
+
+**my_custom_deployer.py**
+
+.. code-block:: python
+
+    def deploy(graph, output_folder: str):
+
+(Note that the arguments are passed as named parameters, so both the ``graph`` and ``output_folder`` names are mandatory)
```

Comment:
We're not saying anything about this `deploy` method in the built-in section. Should we document it before? I mean here:

```
While Conan provides two built-in ones (``full_deploy`` and ``direct_deploy``), users can easily manage their own
with ``conan config install``.
```
Anything to be added there?

Reply:
I see what you mean. The built-in ones are not meant to be customized by the user, so `depoly()` does not apply to them. All the users should see about the built-in deployers is that they are names that can be invoked in the CLI to copy some files around.

What kind of info do you think should be there for them if not?

Reply:
it seems like the name of built-in deployer ([full_deploy](https://github.com/conan-io/conan/blob/develop2/conan/internal/deploy.py#L47)) doesn't match documented one (`deploy()`), so they kinda violate the rule documented?